termux-shared/src/main/java/com/termux/shared/interact/ShareUtils.java

```diff
@@ -0,0 +1,91 @@
+package com.termux.shared.interact;
+
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.core.content.ContextCompat;
+
+import com.termux.shared.R;
+import com.termux.shared.data.DataUtils;
+import com.termux.shared.logger.Logger;
+
+public class ShareUtils {
+
+    private static final String LOG_TAG = "ShareUtils";
+
+    /**
+     * Open the system app chooser that allows the user to select which app to send the intent.
+     *
+     * @param context The context for operations.
+     * @param intent The intent that describes the choices that should be shown.
+     * @param title The title for choose menu.
+     */
+    private static void openSystemAppChooser(final Context context, final Intent intent, final String title) {
+        if (context == null) return;
+
+        final Intent chooserIntent = new Intent(Intent.ACTION_CHOOSER);
+        chooserIntent.putExtra(Intent.EXTRA_INTENT, intent);
+        chooserIntent.putExtra(Intent.EXTRA_TITLE, title);
+        chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        context.startActivity(chooserIntent);
+    }
+
+    /**
+     * Share text.
+     *
+     * @param context The context for operations.
+     * @param subject The subject for sharing.
+     * @param text The text to share.
+     */
+    public static void shareText(final Context context, final String subject, final String text) {
+        if (context == null) return;
+
+        final Intent shareTextIntent = new Intent(Intent.ACTION_SEND);
+        shareTextIntent.setType("text/plain");
+        shareTextIntent.putExtra(Intent.EXTRA_SUBJECT, subject);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;subject&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;context&#39;. The same parameter type is validated here for example: app/src/main/java/com/termux/app/TermuxService.java:510. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).